src/operation.ts

```diff
@@ -0,0 +1,42 @@
+// operation.ts
+
+/*
+ * Copyright (c) 2021-2023 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+export interface CreateOperation<Item> {
+  type: 'create';
+  createdOn: string;
+  item: Item;
+}
+
+export interface DeleteOperation {
+  type: 'delete';
+  createdOn: string;
+  name: string;
+  previousCreatedOn: string;
+}
+
+export type Operation<Item> = CreateOperation<Item> | DeleteOperation;
+
+export function getItemsFromOperations<Item extends { name: string }>(
+  operations: Operation<Item>[],
+  at: string,
+): Item[] {
+  const itemMap = new Map<string, Item>();
+
+  const sortedOperations = operations.sort((operation1, operation2) =>
+    operation1.createdOn.localeCompare(operation2.createdOn),
+  );
+  for (const operation of sortedOperations) {
+    if (operation.type === 'create' && operation.createdOn <= at) {
```

Comment:
i don't think this should be <= but should be <. the end time for balance and entry calculation in ledger are all exclusive on the end because we don't know exactly when in that millisecond the item actually exists (that's why if you create an entry at 2023-10-23T17:26:57.399Z and then try to calculate the balance at that same time, it is not included in the balance calculation). from a practical standpoint it doesn't really make a big difference here because of how frequent these are updated but it should be consistent with the other services we have.